Sources/SwiftBSON/BSON.swift

```diff
@@ -95,6 +95,26 @@ public enum BSON {
             self = int64.bson
             return
         }
+
+        if let uuidJSON = try json.unwrapObject(withKey: "$uuid", keyPath: keyPath) {
+            guard let uuidString = uuidJSON.stringValue else {
+                throw DecodingError._extendedJSONError(
+                    keyPath: keyPath,
+                    debugDescription: "Could not parse `uuid` from \"\(uuidJSON)\", "
+                )
+            }
+            guard let uuid = UUID(uuidString: uuidString) else {
+                throw DecodingError._extendedJSONError(
+                    keyPath: keyPath,
+                    debugDescription: "Invalid UUID string"
+                )
+            }
+
+            let binary = try BSONBinary(from: uuid)
```

Comment:
Because this path is only used in the decoder, we need to repackage any errors that are thrown into `DecodingError`s. We can do that here by wrapping this line in a `do { ... } catch { ... }` that preserves the original error's description. See [here](https://github.com/mongodb/swift-bson/blob/master/Sources/SwiftBSON/BSONBinary.swift#L191) for an example of this.